bundles/org.openhab.binding.jablotron/src/main/java/org/openhab/binding/jablotron/internal/handler/JablotronOasisHandler.java

```diff
@@ -0,0 +1,116 @@
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.jablotron.internal.model.JablotronControlResponse;
+import org.openhab.binding.jablotron.internal.model.JablotronServiceDetailSegment;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link JablotronOasisHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Ondrej Pecta - Initial contribution
+ */
+@NonNullByDefault
+public class JablotronOasisHandler extends JablotronAlarmHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(JablotronOasisHandler.class);
+
+    public JablotronOasisHandler(Thing thing, String alarmName) {
+        super(thing, alarmName);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(CHANNEL_COMMAND) && command instanceof StringType) {
+            scheduler.execute(() -> {
+                sendCommand(command.toString());
+            });
+        }
+
+        if (channelUID.getId().equals(CHANNEL_STATUS_PGX) && command instanceof OnOffType) {
+            scheduler.execute(() -> {
+                controlSection("PGM_1", command.equals(OnOffType.ON) ? "set" : "unset");
+            });
+        }
+
+        if (channelUID.getId().equals(CHANNEL_STATUS_PGY) && command instanceof OnOffType) {
+            scheduler.execute(() -> {
+                controlSection("PGM_2", command.equals(OnOffType.ON) ? "set" : "unset");
+            });
+        }
+    }
+
+    @Override
+    protected void updateSegmentStatus(JablotronServiceDetailSegment segment) {
+        logger.debug("Segment id: {} and status: {}", segment.getSegmentId(), segment.getSegmentState());
+        State newState = "unset".equals(segment.getSegmentState()) ? OnOffType.OFF : OnOffType.ON;
+        switch (segment.getSegmentId()) {
+            case "STATE_1":
+                updateState(CHANNEL_STATUS_A, newState);
+                break;
+            case "STATE_2":
+                updateState(CHANNEL_STATUS_B, newState);
+                break;
+            case "STATE_3":
+                updateState(CHANNEL_STATUS_ABC, newState);
+                break;
+            case "PGM_1":
+                updateState(CHANNEL_STATUS_PGX, newState);
+                break;
+            case "PGM_2":
+                updateState(CHANNEL_STATUS_PGY, newState);
+                break;
+            default:
+                logger.debug("Unknown segment received: {} with state: {}", segment.getSegmentId(),
+                        segment.getSegmentState());
+        }
+    }
+
+    public synchronized void controlSection(String section, String status) {
+        logger.debug("Controlling section: {} with status: {}", section, status);
+        JablotronControlResponse response = sendUserCode(section, section.toLowerCase(), status, "");
+
+        updateAlarmStatus();
```

Comment:
You don't need to poll the state, as the Channel has already the new value, the user set before.

Or does setting one Channel affect the state of another Channel?

Reply:
Yes, sending a user code immediately changes states of the sections. So it is the fastest way how to update the other channels states.